### PR TITLE
addonHandler: Added getTranslatedMessage function.

### PR DESCRIPTION
**Pull request by nvdaes**
Based on the message method created by @ABuffEr, letting add-ons to use messages translated in NVDA's core.

### CODE REVIEW

**Comment by michaelDCurran** [comment]
@nvdaes can you explain why exactly this is needed? For instance, explain why the add-on developer can't just call 
```
_(message)
```

**Comment by nvdaes** [comment]
Hello. When we just call _(message), if the Python file containing the 
message calls addonHandler.initTranslation(), the message is included in 
the nvda.po for the add-on and translators have to translate it.
With getTranslatedMessage, if it's included in NVDA's core, translators:
1. Can avoid to translate the text again.
2. This can make translated messages more consistent avoiding different 
translations for the same message.

One recent example is the word "Error", used in the Emoticons add-on. As 
it's in NVDA's core, we use a sfunction similar to 
getTranslatedMessage("error") instead of _("Error").
Cheers.

File: source/addonHandler.py [hunk 1]
```diff
@@ -437,6 +437,9 @@ def initTranslation():
 	finally:
 		del callerFrame # Avoid reference problems with frames (per python docs)
 
+def getTranslatedMessage(translatedMessage):
```
Comment (michaelDCurran) [inline source/addonHandler.py]: A docstring for this function please.

**Review by michaelDCurran — CHANGES_REQUESTED** [review]

**Comment by nvdaes** [comment]
Done in c75c263
Thanks.

**Comment by jcsteh** [comment]
@MichaelDCurran and I discussed this. One concern we have is that there's no guarantee that a message won't get removed from or changed in core. If that happens, the message in your add-on will be mysteriously untranslated in some versions of NVDA but not others, which could become difficult to manage.

Are there exammples other than "Error" that might be more compelling? Otherwise, I don't think we should go ahead with this for the above reason.

**Comment by nvdaes** [comment]
Other messages can be on, off, etc.
But your concern is true.
Other concern is that messages can be translated in some languages and 
not in others.
Perhaps this should be add-ons territory.
This have advantages and drowbacks.

**Comment by jcsteh** [comment]
Are you happy to close this for now given the above reasoning? Happy to reconsider if there's a really compelling use case, but it seems like a lot of the messages we're talking about here are quite small anyway and are relatively low effort to translate.

**Comment by nvdaes** [comment]
I think you can close this: Your reasoning is good for me.
Anyway, I prefer that translators don't translate messages contained in 
NVDA's core when possible, also for consistency.
For instance in an add-on named placeMarkers, which extends the Find 
dialog in browse mode.
But I agree in your reasoning and for me you can close this.
Thanks.

**Comment by jcsteh** [comment]
Closing as per https://github.com/nvaccess/nvda/pull/6747#issuecomment-273696649. We can reopen this if there's a more compelling case in future. Despite this decision, we do appreciate your work. Thanks.

**Pull request closed without merging**